sw/device/silicon_creator/lib/chip_info.h

```diff
@@ -0,0 +1,26 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_CHIP_INFO_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_CHIP_INFO_H_
+
+#include <assert.h>
+#include <stdint.h>
+
+typedef struct chip_info {
+  uint32_t hw_rev;
+  uint64_t scm_revision;
+  char build_date_str[20];
+} chip_info_t;
+
+// This struct contains information about the ROM's provenance, and critically,
+// it can be manually decoded from a hexdump. The linker script should place
+// this at the top of the ROM.
+extern const chip_info_t kChipInfo __attribute__((section(".chip_info")));
+
+// Set an arbitrary upper bound on the size of `kChipInfo` to prevent future
+// additions from wasting space in the ROM.
+static_assert(sizeof(kChipInfo) <= 64u, "Size of kChipInfo exceeds budget.");
```

Comment:
I guess this bound should be made exact by looking at the top-defined start and end symbols.

Reply:
Abandoning this idea because the constants are defined in the linker script, and it's not clear to me how to align them.